### PR TITLE
[MONDRIAN-2441] fixed calculated member set issue for multicolumn result

### PR DESCRIPTION
**Pull request by andrey-markelov**
I added support for disjoint tuples as expanded element

@mkambol please review it

### CODE REVIEW

File: src/main/mondrian/rolap/SqlConstraintUtils.java [hunk 1]
```diff
@@ -696,43 +746,70 @@ public static boolean hasMultiPositionSlicer(List<Member> slicerMembers) {
         List<Member> members,
         Evaluator evaluator)
     {
-      return expandSupportedCalculatedMembers(members, evaluator, false);
+      return expandSupportedCalculatedMembers(members, evaluator, false, null);
```
Comment (mchen-len-son) [inline src/main/mondrian/rolap/SqlConstraintUtils.java]: Can be just expandSupportedCalculatedMembers(members, evaluator, false), can't it?

File: src/main/mondrian/rolap/RolapEvaluator.java [hunk 2]
```diff
@@ -549,6 +540,20 @@ public boolean isDisjointSlicerTuple() {
         return disjointSlicerTuple;
     }
 
+    public void updateDisjointStatus() {
```
Comment (mchen-len-son) [inline src/main/mondrian/rolap/RolapEvaluator.java]: do we really need it public?

File: src/main/mondrian/rolap/SqlConstraintUtils.java [hunk 3]
```diff
@@ -92,16 +128,28 @@ public static void addContextConstraint(
             // request is impossible to satisfy.
             return;
         }
+
         if (disjointSlicerTuples) {
+            slicerTupleList.add(slicerTuples);
+        }
+
+        if (!disjointSlicerTuples && slicerTupleList.size() > 0) {
+            disjointSlicerTuples = true;
```
Comment (mchen-len-son) [inline src/main/mondrian/rolap/SqlConstraintUtils.java]: this disjointSlicerTuples value is never used further

**Comment by andrey-markelov** [comment]
@mkambol could you review this?

**Comment by andrey-markelov** [comment]
@mkambol @kurtwalker @lucboudreau can you take a look?

**Comment by kurtwalker** [comment]
There is a lot of change in this PR, it'll take me a while to get through it all and I won't have the time to do it today.

**Comment by andrey-markelov** [comment]
@mkambol @kurtwalker @lucboudreau reminder: please review

**Pull request merged**